addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1078,7 +1078,11 @@ var SnippetsMenu = Widget.extend({
                 return;
             }
             this._activateSnippet($target);
-        });
+        };
+
+        this.$document.on('click.snippets_menu', '*', activateSnippetOnClick);
+        // Needed as bootstrap stop the propagation of click events for dropdowns
+        this.$document.on('mouseup.snippets_menu', '.dropdown-toggle', activateSnippetOnClick);
```

Comment:
I don't understand how this code (enabling snippet in the DOM) is related to what your commit message explains (colorpickers in the editor)... I am really confused :grimacing:

Reply:
I realize that I picked the wrong name for the method, it is in charge of activating a snippet if the user clicks on a `we-button`, `we-toggler` or `o_we_color_preview`, but also to close the opened widgets in case of a click somewhere else. But the click event was stopped at the level of the Dropdown because of bootstrap, so to close widgets, a way is to listen to the mouseup event for these particular elements.
I will change the commit title to something more generic as it's not only related to the color picker but to all the widgets.